02_classify_stratified/nbconverted/plot_purity_regression_results.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Plot tumor purity prediction results
+# 
+# Here, we'll visualize the results of our tumor purity prediction. We're predicting real-valued purity labels here (e.g. elastic net regression).
+# 
+# Results analyzed here are generated by the `run_purity_prediction.py` script, without the `--classify` argument.
+
+# In[1]:
+
+
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import matplotlib; matplotlib.rcParams['figure.dpi'] = 300
+import seaborn as sns
+
+import mpmp.config as cfg
+import mpmp.utilities.analysis_utilities as au
+
+
+# In[2]:
+
+
+# set results directory
+results_dir = Path(cfg.results_dir,
+                   'purity_reg_results',
+                   'purity').resolve()
+
+
+# In[3]:
+
+
+results_df = (
+    au.load_purity_results(results_dir)
+      .drop(columns=['identifier'])
+)
+print(results_df.shape)
+print(results_df.training_data.unique())
+results_df.head()
+
+
+# In[4]:
+
+
+sns.set({'figure.figsize': (14, 6)})
+sns.set_style('whitegrid')
+fig, axarr = plt.subplots(1, 2)
+
+plot_df = (
+    results_df[results_df.data_type == 'test']
+      .sort_values(by=['signal', 'training_data'])
+)
+# use 1 - RMSE so higher is better for both metrics
+# all purity values are proportions between 0 and 1, so we're unlikely
+# to have negative 1 - RMSE values
+plot_df['inv_rmse'] = 1 - plot_df.rmse
+
+sns.boxplot(data=plot_df, x='signal', y='inv_rmse', hue='training_data', ax=axarr[0])
+axarr[0].set_title('Tumor purity prediction performance, by data type')
+axarr[0].set_xlabel('Signal or shuffled')
+axarr[0].set_ylabel('1 - RMSE')
+axarr[0].legend(title='Data type')
+sns.boxplot(data=plot_df, x='signal', y='r2', hue='training_data', ax=axarr[1])
+axarr[1].set_title('Tumor purity prediction performance, by data type')
+axarr[1].set_xlabel('Signal or shuffled')
+axarr[1].set_ylabel(r'$R^2$')
+axarr[1].legend(title='Data type')
+
+
+# In[5]:
+
+
+# now facet by cancer type
```

Comment:
Very minor, same as before: I would add `# Facet by cancer type` here, so you can quickly move across the notebook results.

Reply:
I think you showing the top cancer types first, and then all the rest below. I would add section subtitles for those.